fs/unstable_remove_test.ts

```diff
@@ -0,0 +1,153 @@
+  }
+});
+
+Deno.test("remove() remove a non empty directory", async () => {
+  const tempDir1 = await makeTempDir({ prefix: "remove_async_" });
+  const tempDir2 = await makeTempDir({
+    dir: tempDir1,
+    prefix: "remove_async_",
+  });
+  const testFile1 = join(tempDir1, "test.txt");
+  const testFile2 = join(tempDir2, "test.txt");
+
+  const encoder = new TextEncoder();
+  const data = encoder.encode("This is a test content");
+  await writeFile(testFile1, data, { mode: 0o777 });
+  await writeFile(testFile2, data, { mode: 0o777 });
+
+  try {
+    await assertRejects(async () => {
+      await remove(tempDir1);
+    }, Error);
+
+    await remove(tempDir1, { recursive: true });
+    const fsStat = await stat(tempDir1);
+    assert(fsStat.isDirectory === false);
+  } catch {
+    await rm(tempDir1, { recursive: true, force: true });
+  }
+});
+
+Deno.test("remove() remove a non existed directory", async () => {
+  const tempDir = await makeTempDir({ prefix: "remove_async_" });
+  const nonExistedDir = join(tempDir, "non", "existed", "dir");
+
+  try {
+    await assertRejects(async () => {
+      await remove(nonExistedDir);
+    }, NotFound);
+    await remove(tempDir);
+    const fsStat = await stat(tempDir);
+    assert(fsStat.isDirectory === false);
+  } catch {
+    await rm(tempDir, { recursive: true, force: true });
+  }
+});
+
+Deno.test("remove() remove a non existed directory with option", async () => {
+  const tempDir = await makeTempDir({ prefix: "remove_async_" });
+  const nonExistedDir = join(tempDir, "non", "existed", "dir");
+
+  try {
+    await assertRejects(async () => {
+      await remove(nonExistedDir, { recursive: true });
+    }, NotFound);
+    await remove(tempDir);
+    const fsStat = await stat(tempDir);
+    assert(fsStat.isDirectory === false);
+  } catch {
+    await rm(tempDir, { recursive: true, force: true });
+  }
+});
+
+Deno.test("removeSync() remove an existed and empty directory", () => {
+  const tempDir = makeTempDirSync({ prefix: "remove_sync_" });
+  assert(statSync(tempDir).isDirectory === true);
+
+  try {
+    removeSync(tempDir);
+    assert(statSync(tempDir).isDirectory === false);
+  } catch {
+    rmSync(tempDir, { recursive: true, force: true });
+  }
+});
+
+Deno.test("removeSync() remove a non empty directory", () => {
+  const tempDir1 = makeTempDirSync({ prefix: "remove_sync_" });
+  const tempDir2 = makeTempDirSync({
+    dir: tempDir1,
+    prefix: "remove_async_",
+  });
+  const testFile1 = join(tempDir1, "test.txt");
+  const testFile2 = join(tempDir2, "test.txt");
+
+  const encoder = new TextEncoder();
+  const data = encoder.encode("This is a test content");
+  writeFileSync(testFile1, data, { mode: 0o777 });
+  writeFileSync(testFile2, data, { mode: 0o777 });
+
+  try {
+    assertThrows(() => {
+      removeSync(tempDir1);
+    }, Error);
+
+    removeSync(tempDir1, { recursive: true });
+    assert(statSync(tempDir1).isDirectory === false);
+  } catch {
```

Comment:
Ditto. This needs to be `finally`

```suggestion
  } finally {
```

Reply:
Updated.